processor/src/trace/virtual_bus/sum_check/verifier/mod.rs

```diff
@@ -0,0 +1,152 @@
+            eval_domain,
+            final_query_builder,
+            _challenger: PhantomData,
+        }
+    }
+
+    /// Verifies a sum-check proof [Proof] and returns a claim on the openings of the mult-linear
+    /// oracles that are part of the statement being proven.
+    ///
+    /// More precisely, the method:
+    ///
+    /// 1. Generates a `claimed_evaluation` from the round proof polynomials and the round challenge
+    /// randomness.
+    ///
+    /// 2. Computes a query that is built using the [FinalQueryBuilder] from the multi-linear
+    /// openings and the round challenges.
+    ///
+    /// 3. Evaluates the composition polynomial at the query and checks that it is equal
+    /// `claimed_evaluation`.
+    ///
+    /// 4. Outputs a `FinalOpeningClaim` on the multi-linear oracles.
+    ///
+    /// Thus, the proof is correct if the method outputs a [FinalOpeningClaim] and this latter is
+    /// a valid claim on the multi-linear oracles i.e., each multi-linear oracle opens to the
+    /// claimed value at the specified opening point.
+    ///
+    ///  # Errors
+    /// Returns an error if:
+    /// - No openings were provided.
+    /// - Draw the round challenge fails.
+    /// - The final evaluation check fails.
+    pub fn verify(
+        &self,
+        claim: E,
+        proof: Proof<E>,
+        coin: &mut C,
+    ) -> Result<FinalOpeningClaim<E>, Error> {
+        let Proof {
+            openings: openings_claim,
+            round_proofs,
+        } = proof;
+
+        let mut claimed_evaluation = claim;
+        let mut evaluation_point = vec![];
+        for round_proof in round_proofs {
+            let partial_evals = round_proof.poly_evals.clone();
+            coin.reseed(H::hash_elements(&partial_evals));
+            let evals = round_proof.to_evals(claimed_evaluation);
+
+            let r = coin.draw().map_err(|_| Error::FailedToGenerateChallenge)?;
+            let reduced_evaluation = self.eval_domain.evaluate(&evals, r);
+
+            claimed_evaluation = reduced_evaluation;
+            evaluation_point.push(r);
+        }
+
+        if let Some(openings_claim) = openings_claim {
+            if openings_claim.evaluation_point != evaluation_point {
+                return Err(Error::WrongOpeningPoint);
+            }
+            let query = self.final_query_builder.build_query(&openings_claim, &evaluation_point);
+            if self.composition_poly.evaluate(&query) != claimed_evaluation {
+                Err(Error::FinalEvaluationCheckFailed)
+            } else {
+                Ok(openings_claim)
+            }
+        } else {
+            Err(Error::NoOpeningsProvided)
+        }
+    }
+}
+
+/// Contains the logic for building the final query made to the virtual polynomial.
+///
+/// During the last step of the sum-check protocol, the Verifier must evaluate the virtual
+/// polynomial at a random point `(r_0, ... ,r_{\nu - 1})`. To do this, the Verifier asks the Prover
```

Comment:
The term "virtual polynomial" doesn't appear in the `SumCheckProver` docstring that we use as central point of documentation of the algorithm. Could we clearly document what is meant by "virtual polynomial" there?

Reply:
Updated